### PR TITLE
Implement InitUser + minor changes

### PR DESCRIPTION
**Pull request by TomFanella4**
## Changes
1. `InitUser` now checks if the user profile is already created before updating the profile
1. Rename `liprofile` to `profile` (plus similar namings) per previous meetings
1. Added `Greeting` and `ShareLocation` fields to `Profile`
1. Added client constants for `ServerURI` and `chatMessagesToLoad`

### CODE REVIEW

**Review by mpace965 — APPROVED** [review]
Looks good to me. Verified on my machine. Also added some extra logging in case that Firebase error happens again.

File: backend/UserManagement.go [hunk 1]
```diff
@@ -91,17 +92,18 @@ type LiProfile struct {
...
 }
 
 // User is user on MeetOver
 type User struct {
-	ID           string         `json:"uid,omitempty"`
-	Location     *Geolocation   `json:"location,omitempty"`
-	AccessToken  ATokenResponse `json:"accessToken"`
-	LiProfile    LiProfile      `json:"li_profile"`
-	IsSearching  bool           `json:"profile"`
-	HelloMessage string         `json:"hello_message"` // TODO: ask user to fill this feild in account creation
```
Comment (raokrutarth) [inline backend/UserManagement.go:104]: are we not having a greeting at the user profile? That is what hello_message was

Comment (mpace965) [reply backend/UserManagement.go:104]: It got moved to the profile object.

**Comment by TomFanella4** [comment]
Also added `userExists` to the auth response. This is used to tell the client whether the user needs to finish creating their profile or not.

**Pull request merged**